bgpd/bgp_route.c

```diff
@@ -12346,7 +12378,9 @@ DEFUN (show_ip_bgp_instance_neighbor_advertised_route,
 	if (argv_find(argv, argc, "route-map", &idx))
 		rmap_name = argv[++idx]->arg;
 
-	return peer_adj_routes(vty, peer, afi, safi, type, rmap_name, uj);
+	wide = argv_find(argv, argc, "wide", &idx);
```

Comment:
This will also match if the VRF name is `wide` or the peer-group name is `wide`.  Would be better to use `DEFPY` here.

Reply:
replaced DEFUN with DEFPY

Reply:
sorry, I wasn't clear, with DEFPY the parameter passing is more automatic:

- change ` [json | wide]` to ` [json | wide$wide]`
- remove `bool wide = false;`
- remove `wide = argv_find(...)`

Reply:
(DEFPY will create `wide` as a parameter when you use `wide$wide`)

Reply:
Ok thank you .will update the code as per your inputs. 
is this change/ applicable only for "show_ip_bgp_instance_neighbor_advertised_route" ? or even with
. show_ip_bgp_json and
. show_ip_bgp_instance_all ?

Reply:
i mean, shall i use DEFPY(above changes that you mentioned) for the below commands as well 
show_ip_bgp_json &
show_ip_bgp_instance_all

Reply:
I think we could continue with DEFUN here because same applies to json or other parsed occurrences. But let's wait for what @eqvinox says.

Reply:
ok

Reply:
There's no reason to not use `DEFPY` really; at this point we're rolling it out whereever it's useful.  There's already other uses of `DEFPY` in `bgp_route.c` too.  I'd say just use it in all your changes.

Reply:
Ok

Reply:
shall i do the same changes for json as well in these functions? or leave the json as it is ? i mean 

[ json$uj | wide$wide ] &
remove bool uj = use_json(argc, argv);

Reply:
> shall i do the same changes for json as well in these functions?

Sure, that would be bonus add-on cleanup :)

Note you need to change `uj` to `json` then in the code – or (maybe better) use `json$uj` (the part after the `$` is the name for the function parameter.)

There is documentation for this btw:
http://docs.frrouting.org/projects/dev-guide/en/latest/cli.html
(see under "Variable Names" and "DEFPY")

Reply:
Ok. thanks for your inputs and sharing the link :)